### PR TITLE
RHIROS-404 - Enhance messages for new rules

### PR DESCRIPTION
**Pull request by pgarciaq**
Per RHIROS-379, enhance messages to show the details on the issues identified by the new rules for GA, and the list of suggested instances.

This PR introduces two new variables: system.summary and system.candidate_instance_types.

Please note this PR is essentially pseudocode. I have no idea what the actual names of the variables coming from the Advisor rule engine are but it's easier for me to communicate in code than in Google Docs %)

### CODE REVIEW

**Comment by apuntamb** [comment]
Thanks @pgarciaq for enhancing the suggestions. I would like to keep this PR on hold until we have the new rules MR merged to see if the variable names used here are accessible based on the incoming message.

File: seed.d/rules.json
Comment (kgaikwad) [inline seed.d/rules.json:12]: Note that currently these keys are not accessible directly through `system` object.

Comment (apuntamb) [reply seed.d/rules.json:12]: Yes, `instance_type` key could be accessed because it is a column in `system` table.

File: seed.d/rules.json
Comment (apuntamb) [inline seed.d/rules.json:31]: Need to confirm if we do have `Storage Rightsizing` rule key in Advisor engine.

File: seed.d/rules.json
Comment (kgaikwad) [inline seed.d/rules.json:5]: Hello @r14chandra, as ticket [RHIROS-424](https://issues.redhat.com/browse/RHIROS-424) assigned to you. Please note that here it uses `cloud_region` in rules content.
With current implementation, ROS backend don't have `region` information stored in ROS system schema. You will need to consider this as well while working on RHIROS-424.

cc @pgarciaq

Comment (kgaikwad) [reply seed.d/rules.json:5]: @r14chandra - One can fetch `region` information similarly like ros-backend is fetching `instance_type` information [here](https://github.com/RedHatInsights/ros-backend/blob/main/ros/processor/process_archive.py#L48).
Example -  `aws_instance_id.get('region') `

Comment (r14chandra) [reply seed.d/rules.json:5]: Thansk @kgaikwad. Looking into it.

File: seed.d/rules.json
Comment (kgaikwad) [inline seed.d/rules.json:12]: @pgarciaq , one thing in case of `INSTANCE_OVERSIZED`, in sample output of advisor engine we are not getting `summary` information. Will it ok to skip this part from content.

Comment (kgaikwad) [reply seed.d/rules.json:12]: With [latest](https://gitlab.cee.redhat.com/insights-rules/insights-plugins/-/merge_requests/3810) merge request change on advisor rules, I can see messages were added for `INSTANCE_OVERSIZED` state as well. 

https://gitlab.cee.redhat.com/insights-rules/insights-plugins/-/blob/master/telemetry/rules/plugins/ros/ros_instance_evaluation.py#L80

File: seed.d/rules.json
Comment (kgaikwad) [reply seed.d/rules.json:31]: As per offline discussion, currently we don't have required details to compute this state. For future work, we are going to keep state details as it is.

**Comment by PreetiW** [comment]
@kgaikwad @r14chandra Pau have pushed the last bit of changes needed for content to this PR and we can merge this one now. 
You can [refer](https://ansible.slack.com/archives/C01C1GVT3TP/p1639399544354800) this for same.

File: seed.d/rules.json
Comment (kgaikwad) [inline seed.d/rules.json:3]:
```suggestion
        "rule_id": "ros_instance_evaluation|INSTANCE_IDLE",
```

File: seed.d/rules.json
Comment (kgaikwad) [inline seed.d/rules.json:10]:
```suggestion
        "rule_id": "ros_instance_evaluation|INSTANCE_OVERSIZED",
```

**Review by kgaikwad — CHANGES_REQUESTED** [review]
@pgarciaq - added few comments as per sample outputs from jira ticket [CEECBA-5875](https://issues.redhat.com/browse/CEECBA-5875).

**Comment by r14chandra** [comment]
RHIROS-379 (rules content) contains current instance price but this PR doesn't. Which one are we considering?

**Comment by kgaikwad** [comment]
Merging this change as @r14chandra  is working on further improvements in messages using correct data mapping.
@r14chandra  - Please make sure to address unresolved comments in your pull-request https://github.com/RedHatInsights/ros-backend/pull/146.

**Comment by kgaikwad** [comment]
Thank you @pgarciaq for this change!

**Pull request merged**
